### DISCUSSION TITLE
Timeout for Automation

### DISCUSSION THREAD

**Discussion by ConfusedMerlin** [DeviceFarmer/stf#587] [Q&A]
So, imaging having a pipeline (Gitlab) trying to perform an Automation task on a phone in the farm. Trying, because at any point inside the list of tasks gradle performs, it fails for whatever reason. The Gitlab Task itself will switch to failed and its done for this round, but inside the devicefarm there is still the "Stop Automation" button at the list view of the devices for that device.

It will disappear... after some time (I would guess at least thirty minutes), but until then the GitLab have to wait before it can try a rerun, because the Script trying to reconnect detects the "Automation" state and thinks it might be used by another GitLab Pipeline. 

 GitLab itself does have a pending feature request for "execute on failure", which could be used to free up the device in the farm, but... well, it is pending and not sure to be available soon.

Until then, can we actually lower the time the Farm keeps a device in "Automation" when actually no automation happens  at all?

**Comment by denis99999** [comment]
@ConfusedMerlin , you can play on the `--group-timeout` option!
